sys/include/net/nanocoap.h

```diff
@@ -861,6 +861,27 @@ static inline ssize_t coap_get_uri_query_string(coap_pkt_t *pkt, char *target,
                                (uint8_t *)target, max_len, '&');
 }
 
+/**
+ * @brief   Iterate over a packet's URI Query options
+ *
+ * URI Query options will have the format `key=value`.
+ *
+ * @param[in]   pkt           packet to read from
+ * @param[out]  ctx           opaque, must be set to `NULL` on first call
+ * @param[out]  key           Output buffer for the key
+ * @param[in]   key_len_max   Size of the key output buffer
```

Comment:
Why copy out into a buffer in the first place? The get_uri_query function only needed to copy because it is assembling strings, but as this function doesn't guarantee the output to be null terminated, it could just as well populate a `const uint_t **` and a `size_t *`.

Reply:
Either I modify the buffer in the pkt (replace `=` with `\0`) or I have to copy the string if I don't want to leave it to the user to parse it.

Reply:
I agree that modifying the package in-place is not the way to go here.

> It very much does.

Not until it says so in the docs :-)

Reply:
I added it to the doc